### ISSUE TITLE
Bug: automatic loop is only set for songs in the patter directory, not its subdirectories

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#44]
Definitely a bug, not a feature!

**Comment by danlyke** [comment]
Interesting: I think the fix for this might have implications for
tagging: Split the whole path (after the music search path) into
keywords which get added to the keywords, do the search on all of those
things?

And show the last path element (like we currently do) as the name?

**Comment by mpogue2** [comment]
Tags generated from the pathname are a little unusual, and I would argue, they're not obvious and not discoverable.  If we're going to do tags, I'm now thinking that we probably should use some more typical mechanism and UX for this.

I think Thunderbird might be the best model for how to do tags.  Tag menu to add new tag names, same menu to add a tag to a song.  Tags have colors?  In any case, tags should show up in the songTable somewhere, maybe as little icons in a separate Tags column?  

Then, I want the Thunderbird model for column visibility (which I just stuck into a separate issue).

**Comment by mpogue2** [comment]
Oh look, GitHub auto links to the other issue, if I use the "#<number>" format.  Cool.

**Comment by mpogue2** [comment]
I guess what I'm saying is that this is a bug regardless of what we do with Tags...  songs in subdirectories should ideally inherit the Type of the first level directory just after the Music Directory path.

**Comment by mpogue2** [comment]
I think I'm changing my mind on this one.  Support for more than one level of subdirectory is a nice-to-have, but I don't think it's a bug.  The Type is currently the immediate containing directory name, which I now believe is probably correct.  Doing something better than this is an Enhancement, IMHO.

**Comment by mpogue2** [comment]
In any case, if callers feel strongly (as I once did) that this is a Bug, they can speak up! :-)

**Comment by danlyke** [comment]
So revisiting this one, is a reasonable check to see if the one of the identifier strings occurs anywhere in the path name, not just in the final folder?

**Comment by mpogue2** [comment]
I think we have a couple of ways to go on this:

option a) first folder is the type:
<musicDir>/foo/*.mp3 is of type "foo"
<musicDir>/foo/bar/music.mp3 is of type "foo"
<musicDir>/foo/bar/ ... /baz/music.mp3 is of type "foo"

option b) the last containing folder name is the Type, e.g.
<musicDir>/foo/*.mp3 is of type "foo"
<musicDir>/foo/bar/music.mp3 is of type "bar"
<musicDir>/foo/bar/ ... /baz/music.mp3 is of type "baz"

option c) the whole subfolder path is the type
<musicDir>/foo/*.mp3 is of type "foo"
<musicDir>/foo/bar/music.mp3 is of type "foo/bar"
<musicDir>/foo/bar/ ... /baz/music.mp3 is of type "foo/bar/.../baz"

option d) a recognized type anywhere in the path determines the type
<musicDir>/foo/*.mp3 is of type "foo", if "foo" is a known type in Preferences
<musicDir>/foo/bar/music.mp3 is of type "foo" is a known type in Preferences, but it's also of type "bar", and of type "music", and these might conflict.

Given that this was reported by a real user (not me), who expected option a, I suspect that option a is the wiser choice, as per the Principle of Least Surprise.

Options B and C seems to me to be somewhat more powerful and flexible, but they also have more overhead -- when I make a new subdirectory, I also have to remember to add it to the list of Patter types on the Preferences page.

Option D gets a little crazy, because then there's a non 1->1 mapping between paths and types.  I think it's pretty confusing (and surprising).

So, I think the data so far argues for option A), and we possibly revisit it if we get more input from a lot of callers.

**Comment by mpogue2** [comment]
OK,so I went with option A.  Fixed by 7672565cdda0873bd3889e18aba38c174d7f5f92 .
I think this is the one that makes the most sense for now.  It happens that I already had some subdirectories in my Patter and Singing directories, and this made them work pretty well.

I think tagging is a useful thing to add, too, but I think we should do that separately (like Thunderbird), rather than using the file path.
